src/ngraph_graph.h

```diff
@@ -175,19 +176,26 @@ inline std::string get_backend_name(const mxnet::Context &context) {
   }
 }
 
+static std::mutex mx_backend;
 inline std::shared_ptr<ngraph::runtime::Backend> GetBackendFromContext(
   const mxnet::Context &context, bool create = true) {
   auto backend_name = get_backend_name(context);
   auto backend_key = backend_name + ":" + std::to_string(context.dev_id);
-  if (backends.count(backend_key) == 0) {
-    if (!create)
-    {
+  // atomic write access to the static backend weak_ptr
+  std::unique_lock<std::mutex> lock(mx_backend);
+  // default construct a backend weak_ptr, will not lock
+  if (auto backend = backends[backend_key].lock()) {
+    return backend;
```

Comment:
In general, I dislike early returns, I'd prefer to simply return at the end if possible.